protocol/common/message_sender.go

```diff
@@ -782,11 +782,16 @@ func (s *MessageSender) HandleMessages(shhMessage *types.Message) ([]*v1protocol
 				hlogger.Error("failed to handle transport layer message", zap.Error(err))
 				return nil, nil, err
 			}
+			err = s.handleEncryptionLayer(context.Background(), &statusMessage)
+			if err != nil {
+				hlogger.Debug("failed to handle an encryption message", zap.Error(err))
+			}
+
 			stms, as, err := unwrapDatasyncMessage(&statusMessage, s.datasync)
 			if err != nil {
 				hlogger.Debug("failed to handle datasync message", zap.Error(err))
 				//that wasn't a datasync message, so use the original payload
-				statusMessages = append(stms, &statusMessage)
```

Comment:
Ah, so this is the wrong loop variable being used?

Reply:
that's right, it was using the wrong variable